File: javascript/binary/binary.js
class Binary {
  
  constructor(num){
    this.num = num;
  }
  
  toDecimal(){
    if (this.num.match(/[^0-1]/)) return 0;
    
    return this.num.split('').reverse().reduce((arr,x,i) => {
      return arr + x * 2 ** i; 
    },0);
  }
  
}

module.exports = Binary

File: ruby/binary/binary.rb
module BookKeeping
  VERSION = 3
end
  

class Binary
  def self.to_decimal(num)
    raise ArgumentError if num.to_s =~ /[^0-1]/
    num.to_s.chars.reverse.each_with_index.reduce(0) do |r, (x, i)|
      r + x.to_i * 2**i
    end
  end
end
